spec.bs

```diff
@@ -186,44 +186,74 @@ Each {{URLPattern}} object has an associated <dfn for=URLPattern>search componen
 Each {{URLPattern}} object has an associated <dfn for=URLPattern>hash component</dfn>, a [=component=], which must be set upon creation.
 
 <dl class="domintro non-normative">
-  <dt><code>{{URLPattern}} . {{URLPattern/protocol}}</code>
+  <dt><code>|urlPattern| = new {{URLPattern/constructor(input, baseURL)|URLPattern}}(|input|)</code></dt>
   <dd>
-    <p>The normalized protocol pattern string.
+    Constructs a new URLPattern object.  The |input| is an object containing separate patterns for each URL component; e.g. hostname, pathname, etc.  Missing components will default to a wildcard pattern.  In addition, |input| can contain a baseURL property that provides static text patterns for any missing components.
   </dd>
 
-  <dt><code>{{URLPattern}} . {{URLPattern/username}}</code>
+  <dt><code>|urlPattern| = new {{URLPattern/constructor(input, baseURL)|URLPattern}}(|input|, |baseURL|)</code></dt>
   <dd>
-    <p>The normalized username pattern string.
+    Constructs a new URLPattern object.  The |input| is a URL string containing pattern syntax for one or more components.  If |baseURL| is provided, then |input| can be relative.  This constructor will always set at least an empty string value and does not default any components to wildcard patterns.
   </dd>
 
-  <dt><code>{{URLPattern}} . {{URLPattern/password}}</code>
+  <dt><code>|matches| = |urlPattern|.{{URLPattern/test(input, baseURL)|test}}(|input|)</code></dt>
   <dd>
-    <p>The normalized password pattern string.
+    Tests if |urlPattern| matches the given arguments.  The |input| is an object containing strings representing each URL component; e.g. hostname, pathname, etc.  Missing components are treated as empty strings.  In addition, |input| can contain a baseURL property that provides values for any missing components.  If |urlPattern| matches the |input| on a component-by-component basis then true is returned.  Otherwise, false is returned.
   </dd>
 
-  <dt><code>{{URLPattern}} . {{URLPattern/hostname}}</code>
+  <dt><code>|matches| = |urlPattern|.{{URLPattern/test(input, baseURL)|test}}(|input|, |baseURL|)</code></dt>
   <dd>
-    <p>The normalized hostname pattern string.
+    Tests if |urlPattern| matches the given arguments.  The |input| is a URL string.  If |baseURL| is provided, then |input| can be relative.  If |urlPattern| matches the |input| on a component-by-component basis then true is returned.  Otherwise, false is returned.
```

Comment:
```suggestion
    Tests if |urlPattern| matches the given arguments.  |url| is a URL string.  If |baseURL| is provided, then |url| can be relative.
    
    If |urlPattern| matches the |input| on a component-by-component basis then true is returned.  Otherwise, false is returned.
```